### DISCUSSION TITLE
Importance of Grounded Power Supplies

### DISCUSSION THREAD

**Discussion by goobius** [mcguirepr89/BirdNET-Pi#625] [General]
Hi all!

I recently jumped into the Birdnet Pi world to monitor for owls in our area as we are under wildlife management for owls, squirrels, and bats. I am using this data to document when the owls start arriving in our area. Oddly enough, my monitor station detected a Whiskered Screech Owl which is only found in Mexico and rarely in New Mexico!

I am using an RPi4 and the USB-C power supply recommended for it, but it only has 2 prongs and no third ground prong. I did some analysis on a basic electret microphone using Friture and noticed a harmonic starting at roughly 60Hz. Definitely some electrical noise. Tried shielding and toroid's, but no reduction. I popped in a USB-C charger for a laptop that does have all three prongs and it was greatly reduced! Below is a picture of the comparison. The left side is the grounded charger and the right side is the ungrounded one recommended for RPi systems. 

It is definitely worth getting a good quality power supply. My next test is likely with a grounded power strip that also has USB charger ports. 

Keith
<img width="491" alt="AudioSpectrum" src="https://user-images.githubusercontent.com/30322148/198118138-dca17b38-2fc4-4db0-8f2b-b4a9d53e5a2e.png">

**Comment by matthew73210** [comment]
Hey,

Sadly it's not as simple as having a good grounded supply. Having ground can actually make things worse i.e common mode interference where the AC signal will go to ground through capacitive coupling then come around through the earth connection. It usually happens on hifi amps where you hear a massive hum and have to remove the earth.
You can also get differential mode interference where current is making a round trip through the power supply through capacitive coupling. Sounds a bit like what you are experiencing.

Here are some things that you can do, add a choke to the mic line + power supply. You must make sure both cables go though the choke. Basicly what happens is that the field created on one line opposes through the choke current on the other line. How far is the mic away from the rpi because you may also be picking up interference through the mic line? You can also add a ferite bead to one line to remove differential mode interference.
Have a quick look at [https://www.murata.com/~/media/webrenewal/products/emc/emifil/knowhow/26to30.ashx](Differential and Common Mode Noise)
If you run the RPi on a battery pack do you still pickup said interference?

btw I'm an electrical engineer who works for the sncf. I often have to deal with such problems (signaling...). Part of my training at uni was this soft of stuff. It boils down to mitigation because everything is a capacitor for AC. And a capacitor is (depending on Hz) a low impedance path for AC.

Matthew

**Reply by jmtuts** [reply]
Matthew - I get a snapping or popping noise on my setup, I thought it might be a grounding issue as I only have a 2 pronged plug for my rpi.  If a choke or ferite bead are potential solutions can you recommend some? Here is an image of what is happening.



![Screenshot 2023-01-26 114126](https://user-images.githubusercontent.com/123665555/214922147-7b04db55-f9d9-4998-9dc8-424eb51ed7c4.jpg)

John

**Reply by matthew73210** [reply]
Hello :)
You first will need to isolate the problem, do you get the same noises from the mic while plugged into another computer? Are their any mains power lines near the mic? Out of interest how old is the setup? I've had crackling sounds come from a failing cap.
As for recommendations, i'd go for the simplest / cheapest one you can find. 
As @nilspupils mentioned further on, the best bet is to go with a digital mic and use the USB connection. The ADC on the rpi isn't the greatest and is prove to interference.
Cheers

**Reply by jmtuts** [reply]
I have tested the mic on several different computers, the issue does not recur.  My mic sits next to a camera, but I disconnected that to see if it was causing the issue, it was not. The pi is several years old, the mic is a couple of months old.  The mic is using the USB connection, Ihave tried all 4 ports and receive the same sounds

**Reply by matthew73210** [reply]
> I have tested the mic on several different computers, the issue does not recur. My mic sits next to a camera, but I disconnected that to see if it was causing the issue, it was not. The pi is several years old, the mic is a couple of months old. The mic is using the USB connection, Ihave tried all 4 ports and receive the same sounds

If the rpi is 7 years old and you have the same issue on all ports, but not on other computers. I'd quite comfortably say that the rpi is at fault, sounds like a dying regulator or a bad cap on the board which is adding interference onto the mic power. The only option here is to replace the rpi or do an rtsp to the rpi from another source.

**Comment by nilspupils** [comment]
Hi @goobius and @matthew73210, 

i was struggling with noise on my own mics for months and have written about it in the microphone thread. 
I also used a grounded power supply for the pi with no change in the noise. 
What helped me getting noise free were digital mics (Boya-M3) or the use of a proper audio-interface with phantom powered mics (and balanced connections). 
Good luck, 

Nils

**Comment by Vertikar** [comment]
I'm still using the standard RPi USB-C power supply on an RPi4 without any issues, but have had to move the microphone cable a few times to get it away from noise sources.

**Comment by DD4WH** [comment]
The spectrogram shows popping or crackling sound sparks occruing irregularly.

@jmtuts   do you use a RPi 3B+ ?

This crackling noise is exactly what I experience with Raspberry Pi 3B+. If I connect the same mic and the same USB sound card to a RPi4 or RPi 3B or RPi Zero 2W, the crackling noise is gone. So my suspicion would be that the RPi 3B+ produces this and I recommend to use a different RPi model. I have no idea where this phenomenon comes from and how one could fix it . . .

**Reply by jmtuts** [reply]
Thank you all for your input.  Looks like I need a replacement, if I can find one!

**Reply by jmtuts** [reply]
I am up and running with the Libre AML-S905X-CC (Le Potato). A big thank to all who provided info and guidance. No more pops or snaps, just clean data. Same power supply BTW, so I suspect my Rpi was the issue, it has been repurposed to ADS-B work for FlightAware

**Comment by goobius** [comment]
John,

I have a separate post about this board working with Birdnet Pi https://www.amazon.com/dp/B074P6BNGZ?ref=nb_sb_ss_w_as-reorder-t1_ypp_rep_k4_1_6&amp=&crid=2H52N52M94WQ2&sprefix=libre%2B&th=1

Someone else has tried it and it seems to be working for him as well. They are $35 BUT it does not have any onboard wireless or Bluetooth. Mine at the office is on a wired network and has been stable for months. 

This is the other discussion: https://github.com/mcguirepr89/BirdNET-Pi/discussions/642

Keith

**Comment by jmtuts** [comment]
goobius - Thank you for the info, I was unaware of this little gem.  I also assume it doesn't have an onboard sound card as well?

So, for $35, plus a WiFi card and sound card we are still WAY less than a rpi

John

**Reply by goobius** [reply]
I am using a Sabrent USB sound card for this unit. And YES the cost is way less than RPi on the black market right now!

The only ODD part is the GPIO pins are not "mapped" but can be with a wires tool, no clue about that. It does work great for this project. 

Keith

**Comment by Vertikar** [comment]
An update on this from my setup - RPi 4, powered by generic PoE splitter, UGreen mic input with a lavalier mic on about 2 metres of cable.
Not exactly related but maybe a handy hint in the future for someone.

Was getting a random noise being introduced ever 3-4 seconds, had no idea what it was, tried moving the cables further away and all that.
In the end a reboot sorted it out.
